blob/gcsblob/gcsblob.go

```diff
@@ -37,13 +49,57 @@ import (
 	"github.com/google/go-cloud/gcp"
 
 	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2/google"
 	"google.golang.org/api/googleapi"
 	"google.golang.org/api/iterator"
 	"google.golang.org/api/option"
 )
 
 const defaultPageSize = 1000
 
+func init() {
+	blob.Register("gs", func(ctx context.Context, u *url.URL) (driver.Bucket, error) {
+		q := u.Query()
+		opts := &Options{}
+
+		if accessID := q["access_id"]; len(accessID) > 0 {
+			opts.GoogleAccessID = accessID[0]
+		}
+
+		if keyPath := q["private_key_path"]; len(keyPath) > 0 {
+			pk, err := ioutil.ReadFile(keyPath[0])
+			if err != nil {
+				return nil, fmt.Errorf("failed to read private key: %v", err)
+			}
+			opts.PrivateKey = pk
+		}
+
+		var creds *google.Credentials
+		if credPath := q["cred_path"]; len(credPath) == 0 {
+			var err error
+			creds, err = gcp.DefaultCredentials(ctx)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			jsonCreds, err := ioutil.ReadFile(credPath[0])
```

Comment:
This is going to come up a lot. Consider factoring it out to an internal function.

See https://code.googlesource.com/google-api-go-client/+/master/internal/creds.go.